packages/core/strapi/lib/services/entity-service/validation/relations.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const { ApplicationError } = require('@strapi/utils').errors;
+const { castArray } = require('lodash/fp');
+const { isEmpty, uniqBy } = require('lodash');
+
+/**
+ * Check that all the relations of an entity exist
+ */
+module.exports = async (data, uid, db) => {
+  const relationsMap = new Map();
+
+  const relationCheck = (data, uid) => {
+    const model = strapi.getModel(uid);
+    if (isEmpty(model)) return;
```

Comment:
If there was no model for the uid I think we would want to return early. Also some of the tests focusing on other parts of the `entity-validator` and `entity-service` have strapi.getModel mocked to return {}

Reply:
Ok I see! However the model should never be empty so I think it's better if it throws when it is empty (meaning something wrong happened and we want to be aware of that). Also I think it's better if there is no custom code to handle the tests, I would say the tests should adapt to the function and not the other way around. The function should not know about how the tests was written. Could you find a way to adapt the tests instead?

Reply:
I kinda agree with you @petersg83. Although I do think that test and code should adapt to each other.
Maybe, if the model shouldn't be undefined/empty (as if we were in TS), we should throw a specific error to let us know when debugging such an error ?
Something like
```
if (isEmpty(model)) {
   throw new Error("The model must not be empty. There must be an error before calling this function.");
}
```
A way for us to identify clearly what happen (instead of generic error like `TypeError: Cannot read properties of undefined`)

Reply:
I don't think it would be relevant as it doesn't give more useful information (the stacktrace leads us directly to the problem) and it makes the code longer/harder to read. It is an internal function so I don't think it would useful for us to add that line. It very common to get the schema from a uid, if we start handling this level of "errors" then we will have a huge overhead of error handling code